### ISSUE TITLE
 how to clear the room ?

### ISSUE THREAD

**Issue by wendux** [socketio/socket.io#3042]
like  io.to("chat").clear() ,  all socket  in room "chat"  will leave

**Comment by sunstan** [comment]
You can create a listener and a function to do that ;)

```
// Server side
io.to("chat").emit("clear_chat");

// Client side
socket.on("clear_chat", function(){
      // If you use jquery
      $(".message").remove();
});
```

**Comment by rllalloshi** [comment]
Rooms are created and deleted automaticly. That means when the room is empty, there are no more clients left on it, it will be deleted.
You can do this to remove all clients from a room: 


```
io.of('/').in('chat').clients((error, socketIds) => {
  if (error) throw error;

  socketIds.forEach(socketId => io.sockets.sockets[socketId].leave('chat'));

});

```
[https://socket.io/docs/server-api/#namespace-clients-callback](https://socket.io/docs/server-api/#namespace-clients-callback)

**Comment by darrachequesne** [comment]
To complete @rllalloshi's answer, in a multi-server configuration you should use the `removeLeave` method of the adapter:

```js
io.of('/').in('chat').clients((error, socketIds) => {
  if (error) throw error;

  socketIds.forEach(socketId => io.of('/').adapter.remoteLeave(socketId, 'chat'));

});
```

[Reference](https://github.com/socketio/socket.io-redis#redisadapterremoteleaveidstring-roomstring-fnfunction)

**Comment by zeiontic** [comment]
Why is there no simple call to remove all sockets from a room? It is a very common use case

**Comment by ZoeDreams** [comment]
i vote to add. room.kick() function where you can pass in a specific id, an array, or 'ALL" for removing everyone

**Comment by Rahulm2310** [comment]
Is someone working on this ?

**Comment by Rahulm2310** [comment]
> @wendux rooms are created and deleted automatically. That means when the room is empty, there are no more clients left on it, it will be deleted.
> You can do this to remove all clients from a room:
> 
> ```
> io.of('/').in('chat').clients((error, socketIds) => {
>   if (error) throw error;
> 
>   socketIds.forEach(socketId => io.sockets.sockets[socketId].leave('chat'));
> 
> });
> ```
> 
> https://socket.io/docs/server-api/#namespace-clients-callback
@rllalloshi didn't worked me
TypeError: io.of(...).in(...).clients is not a function

**Comment by darrachequesne** [comment]
For future readers: this was implemented by https://github.com/socketio/socket.io/commit/b25495c069031674da08e19aed68922c7c7a0e28 and included in `socket.io@4.0.0`.

Syntax:

```js
// make all Socket instances leave the "room1" room
io.socketsLeave("room1");

// make all Socket instances of the "admin" namespace in the "room1" room leave the "room2" room
io.of("/admin").in("room1").socketsLeave("room2");
```

Documentation: https://socket.io/docs/v3/migrating-from-3-x-to-4-0/#Additional-utility-methods

**Comment by taylordwright44** [comment]
> For future readers: this was implemented by [b25495c](https://github.com/socketio/socket.io/commit/b25495c069031674da08e19aed68922c7c7a0e28) and included in `socket.io@4.0.0`.
> 
> Syntax:
> 
> ```js
> // make all Socket instances leave the "room1" room
> io.socketsLeave("room1");
> 
> // make all Socket instances of the "admin" namespace in the "room1" room leave the "room2" room
> io.of("/admin").in("room1").socketsLeave("room2");
> ```
> 
> Documentation: https://socket.io/docs/v3/migrating-from-3-x-to-4-0/#Additional-utility-methods

So how do I remove a specific ID using that? I can't seem to find any documentation on removing just a single ID in v4. The way I used for socket.io 2 doesn't work either.

**Comment by darrachequesne** [comment]
@taylordwright44 since the socket joins the room identified by its own ID, you can use:

```js
io.in(theSocketId).socketsLeave("room1");
```

I've added this example in the documentation. Thanks :+1:
